launcher/modplatform/flame/FileResolvingTask.cpp

```diff
@@ -153,7 +154,7 @@ void Flame::FileResolvingTask::modrinthCheckFinished()
         // If there's more than one mod loader for this version, we can't know for sure
         // which file is relative to each loader, so it's best to not use any one and
         // let the user download it manually.
-        if (file.loaders.size() <= 1) {
+        if (std::bitset<8>(file.loaders).count() <= 1) {
```

Comment:
You name it `low level` but here we are dealing with bit maps(ModloaderTypes is in fact a bitset). 
So to determine if this version has only one loader we have the following options:
- the current one 
- using [std::has_single_bit](https://en.cppreference.com/w/cpp/numeric/has_single_bit) (did not try it and I'm not sure if it will build on all architectures)
- using [std::popcount(x) == 1](http://en.cppreference.com/w/cpp/numeric/popcount) (did not test it and may not build on all architectures)
- using a for loop over all known values of ModLoaderTypes and counting them (I would pretty much want to avoid this one)
- using this complicated bytewise operation: `file.loaders && !(file.loaders & (file.loaders - 1))` (this is fine but on my opinion is more low level than the current approach)

These are all my suggestions regarding how to implement this.
Can you please suggest what should I use? Also if you know a better solution I'm open to it.

Reply:
Maybe low-level wasn't the right way to put it.
I think the last option would be more reliable, as afaik `8` would need to change at some point if more loaders were added.

Reply:
Ok, then are you ok with has_single_bit?

Reply:
I believe our target is C++17 so that won't work.

Reply:
Then the last option :) : `file.loaders && !(file.loaders & (file.loaders - 1))`

Reply:
:woozy_face:

Reply:
Why are you including `<bit>`?

Reply:
I did try the first to options before going to the last one. Removed it